packages/drivers/routerlicious-driver/src/documentStorageService.ts

```diff
@@ -64,15 +66,17 @@ export class DocumentStorageService implements IDocumentStorageService {
             requestVersion = versions[0];
         }
 
-        const rawTree = await this.manager.getTree(requestVersion.treeId);
+        const rawTree = await this.manager.getTree(requestVersion.treeId)
+            .catch(DocumentStorageService.enhanceGitServiceError);
         const tree = buildHierarchy(rawTree, this.blobsShaCache);
 
         this._logTailSha = ".logTail" in tree.trees ? tree.trees[".logTail"].blobs.logTail : undefined;
         return tree;
     }
 
     public async getVersions(versionId: string, count: number): Promise<IVersion[]> {
-        const commits = await this.manager.getCommits(versionId ? versionId : this.id, count);
+        const commits = await this.manager.getCommits(versionId ? versionId : this.id, count)
```

Comment:
I'm not sure I like the way error handling is done. There is really nothing preventing IGitManager being implemented differently (i.e. having alternative implementations) and have error reporting done in different way / structure. I feel like conversion to "right" format should happen as close to network call as possible, i.e. stay at the same layer.
Also not very clear where / how you are going to add retries for retryable errors (offline, 429, general errors, as well as expired token that requires token refresh).
Will it be at same layer? At IGitManager  layer?